core/modules/admin_bar/admin_bar.theme.inc

```diff
@@ -73,7 +73,10 @@ function theme_admin_bar_links($variables) {
     $icon = '';
     if ($elements[$path]['#level'] < 2) {
       if (isset($elements[$path]['#options']['icon'])) {
-        $icon = icon($elements[$path]['#options']['icon'], array('immutable' => TRUE));
+        // If 'icon' is set to an empty string or FALSE, do not add an icon.
+        if ($elements[$path]['#options']['icon']) {
+          $icon = icon($elements[$path]['#options']['icon'], array('immutable' => TRUE));
```

Comment:
I wonder, if this "immutable" is actually necessary.

It may apply to other items and it may be, that someone wants to add an icon later... :thinking:

Reply:
Yeah, I did this so that themes could provide their own versions of icons but have it not affect the admin bar, since otherwise the icon would change between front-end and admin- themes.

An alternative for those really looking to change the icons is they could override the admin bar output, either with `hook_menu_alter()` and change the icon in the menu, or `hook_admin_bar_output_alter()` and change the icon option.